[docs] add GitLab importer docs (#44295)
gravitational/teleport: docs/pages/access-controls/access-graph/gitlab.mdx

---
title: Discover GitLab Access Patterns with Teleport Policy
description: Describes how to syncronize GitLab access patterns using Teleport Policy and Access Graph.
---

> Teleport Policy streamlines and centralizes access management across your entire infrastructure, allowing you to view
> access relationships within seconds. You can see unified, up-to-date relationships and policies between all users, groups,
> and computing resources.

Reviewer: Nitpick but I would remove this paragraph. The less "happy talk" the better. The next one reads like a very good intro already.
```suggestion
```

With Teleport Policy's Access Graph, you gain insights into access patterns within your GitLab account. By scanning all
permissions, users, groups, and projects, it provides a visual representation to help enhance the permission model within
your GitLab environment. This functionality enables you to answer queries such as:

- What projects are accessible to users?
- Which users have write permissions to projects?

Access Graph is a feature of the [Teleport Policy](https://goteleport.com/platform/policy/) product

> available to Teleport Enterprise customers.

Reviewer: ```suggestion
available to Teleport Enterprise and Teleport Cloud customers.
```

Author: Our docs linter denies using `Teleport Enterprise and Teleport Cloud`, instead it suggests `Teleport Enterprise edition`

After logging in to the Teleport UI, navigate to the Management tab. If enabled, Access Graph options can be found
under the Permission Management section.

## How it works

Access Graph synchronizes various GitLab resources, including users, projects and groups.
These resources are then visualized using the graph representation detailed in the
[Access Graph page](../teleport-policy/getting-started-policy.mdx).

The importing process involves two primary steps:

### Polling GitLab APIs

The Teleport cluster continuously scans the configured GitLab accounts and retrieves the following resources:

- Users
- Groups
- Projects
- Group memberships
- Project memberships

Once all the necessary resources are fetched, Teleport pushes them to the
Access Graph, ensuring that it remains updated with the latest information from your GitLab instance.

### Importing resources

Teleport Policy’s Access Graph feature delves into the resources imported and their relationships, crafting a
graphical representation thereof.

> ## Prerequisites

Reviewer 2: Do we know what is the lowest supported GitLab self-hosted version?

Author: Gitlab v4 was introduced in Gitlab version 6 and v3 was removed in Gitlab 9.

- A running Teleport Enterprise cluster v14.3.20/v15.3.1/v16.0.0 or later.
- For self-hosted clusters, an updated `license.pem` with Teleport Policy enabled.

> - For self-hosted clusters, a running Access Graph node v1.20.0 or later.

Reviewer 2: I'd suggest the latest as then we added a few UI fixes to the integration.
```suggestion
- For self-hosted clusters, a running Access Graph node v1.21.4 or later.
```

Check [Access Graph page](../teleport-policy/getting-started-policy.mdx) for details on
how to set up Access Graph.
- The node running the Access Graph service must be reachable

> from Teleport Auth Service.

Reviewer: We should make it clear somehow that it's only for self-hosted and if you're using Cloud you don't have to do anything other than purchase it (unless it's a trial).

</Notice>

The token will be used in the next step to configure the GitLab Sync integration.

## Step 2/3. Set up Access Graph GitLab Sync

To initiate the setup wizard for configuring GitLab Sync, access the Teleport UI,
navigate to the Management tab, and choose the Access Graph option within the
Permission Management section.

In the Access Graph page, you'll notice a button labeled `Integrations`. Click on it to
to access the Integrations page. On the Integrations page, click on the `Setup` button next to the GitLab integration.

You'll be prompted to provide the GitLab token created in Step 1 and the GitLab instance domain.
Once the token is successfully validated, you'll be able to see the resources imported in Access Graph.

## Step 3/3. View GitLab resources in Access Graph

After the GitLab resources are imported, you can view them in the Access Graph page.
The graph representation will show the relationships between users, groups, and projects within your GitLab instance.

Users can have permissions to access a Group or Project. When a user has access to a Group, they inherit permissions
to all projects and subprojects within that group.

> 

Reviewer: Can we give a few examples of queries users can run to analyze Gitlab access paths?
